.github/workflows/docusaurus.yml

```
@@ -0,0 +1,44 @@
+name: docusaurus
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    name: Build Docusaurus
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 18
+          cache: "yarn"
+          cache-dependency-path: "app"
+
+      - name: Set up Yarn
+        working-directory: ./doc
+        run: |
+          corepack enable
+          corepack prepare yarn@4.3.1 --activate
```

Fast-fail if `corepack` fails
```suggestion
        run: corepack enable && corepack prepare yarn@4.3.1 --activate
```